src/Compilers/CSharp/Test/Semantic/Semantics/OverloadResolutionPerfTests.cs

```diff
@@ -987,5 +987,231 @@ static void M(object o)
             var containingTypes = exprs.SelectAsArray(e => model.GetSymbolInfo(e).Symbol.ContainingSymbol).ToTestDisplayStrings();
             Assert.Equal(new[] { "A", "B", "B", "A", "B", "B" }, containingTypes);
         }
+
+        [Fact]
+        public void NestedLambdasAndOverloads_NoTypes_NoErrors()
```

Comment:
I think we should verify the IOperation trees for these tests to make sure that we know what semantic information will be availble to the IDE in these cases.